### DISCUSSION TITLE
Leap second correction

### DISCUSSION THREAD

**Discussion by Macintosh-Fan** [jpb10/SolarCalculator#18] [Q&A] [answered]
Just curious:
With leap seconds, an extra second is added, but the earth rotation does not stop. So, for the sun calculator, does it take the leap seconds that have occurred before into account?

**Comment by jpb10** [comment, marked as answer]
The leap second is well within the margin of error for these calculations (accuracy of about 1 arcminute).

In fact, [delta T](https://en.wikipedia.org/wiki/%CE%94T_(timekeeping)), the time difference between terrestrial time and universal time (currently about 70 seconds), is not taken into account.

**Comment by jpb10** [comment]
As I learned more about timing over the past year, I remembered that I didn't really answer this question correctly. 

This library assumes that [Coordinated Universal Time](https://en.wikipedia.org/wiki/Coordinated_Universal_Time) (UTC) is the same as [Universal Time](https://en.wikipedia.org/wiki/Universal_Time) (UT or UT1). Currently, UTC is kept within ±0.9 second of UT1 by adding leap seconds to it.

If you wanted to maintain a continuous solar position when leap seconds are added, then you could use UT1 instead of UTC as a time input. But that's not necessary, in my opinion, since the maximum discrepancy is < 0.004 degrees (at 0.9 sec) and on average only 0.002 degrees at any given time.
